packages/block-editor/src/components/rich-text/native/index.native.js

```diff
@@ -336,7 +336,7 @@ export class RichText extends Component {
 		);
 
 		this.debounceCreateUndoLevel();
-		const refresh = this.value !== contentWithoutRootTag;
```

Comment:
This is the specific line resulting in incorrectly clearing pending redo actions. Comparing a `RichTextData` instance (`this.value`) against a string (`contentWithoutRootTag`) never resulted in equality. Making the situation more complex is the fact that iOS and Android behave differently when it comes to blur and selection events, which means this nuanced scenario only occurred on iOS. 

The rest of the added `toString` invocations separate of this line in the proposed changes are out of caution, not to address an explicitly observed issue. Transparently, it is unclear to me currently when/where `RichTextData` _should_ be leverage and where Aztec's string-value-based communication might disrupt that by passing back a string. It could be that a better solution is to always leverage `RichTextData` values in JavaScript and convert values received from Aztec from string to `RichTextData`.

Reply:
To be clear, the use of `toString` is fairly innocuous as both `String` and `RichTextData` provide the method. So, regardless of which type is present, the method and comparison should succeed. 

If nothing else, we could use the proposed changes as a stopgap solution until we fully integrate the `RichTextData` type.